Add dispute details screen

dechov opened the pull request:
Fixes #350 

#### Changes proposed in this Pull Request

Add a "Dispute Details" screen with a Challenge button linking to the evidence form:

<img width="1145" alt="Screen Shot 2020-02-10 at 1 02 55 AM" src="https://user-images.githubusercontent.com/1867547/74124684-19a0ae00-4ba1-11ea-8cdb-4fc740b588ea.png">

…linked via a new info button in Disputes list rows:

<img width="1067" alt="Screen Shot 2020-02-10 at 12 34 59 AM" src="https://user-images.githubusercontent.com/1867547/74123587-536fb580-4b9d-11ea-8cb6-d4cb06985418.png">

The copy for the details page is derived from a google doc shared by @LevinMedia.

Also:
- removes small placeholder on info button in Transactions and Disputes lists
- removes link from "Needs Response" status chip to evidence form (just added in https://github.com/Automattic/woocommerce-payments/pull/410)

Treating container width and outsize vertical padding (being discussed in https://github.com/Automattic/woocommerce-payments/pull/414) as out of scope.

The dispute info section near the top is tracked in https://github.com/Automattic/woocommerce-payments/issues/379.

The "Accept Dispute" button can also be a different PR in my view (not currently tracked in its own issue but blocking https://github.com/Automattic/woocommerce-payments/issues/376 and https://github.com/Automattic/woocommerce-payments/issues/377).

Question: should the info button only appear if the dispute needs a response? Or for all rows, as now, except we'll need to disable things / make it clear it's not actionable? (cc @LevinMedia; related: https://github.com/Automattic/woocommerce-payments/issues/379#issuecomment-583965218)

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Open the Disputes list view, and click on the info button to the left
* Verify that appropriate text shows up for disputes with the different statuses

-------------------

- [ ] Tested on mobile (or does not apply)

<!--
Please read P7bbVw-3ww-p2 before opening the PR, assign the correct status to it, __and make sure that the related issue uses the Has PR status!__.
-->

dechov committed:
Initial dispute details route with 'challenge' button

dechov committed:
Info button on disputes list screen

dechov committed:
Dispute response prose conditional on type

dechov committed:
Omit headings for missing sections

dechov committed:
Add dispute details snapshot tests

vbelolapotkov reviewed (changes requested):
@dechov great progress on that! 👍 

I've tested with fraudulent and product not received disputes and confirm different content is shown for each of them. However, I tried to check if messaging matches the gdoc but not sure what's the correct version so leaving this check to the @LevinMedia . I've noticed headings in the message are a bit bigger than card title, not sure that's correct. @LevinMedia will 

<img width="369" alt="Screenshot 2020-02-10 at 13 58 47" src="https://user-images.githubusercontent.com/3139099/74144277-7fc31a80-4c0d-11ea-8c9a-2e38f0f79919.png">


Some general notes:
1. Noticed that one test snapshot is out of date, see `npm test` output for details. 
2. Phpunit tests are failing as well but the reason seems to be unrelated to the current changes.

See other notes and comments below.

dechov committed:
Move Paragraphs into components directory, and fix/enhance snapshot tests

dechov committed:
Update dispute details and evidence styles to match designs

dechov committed:
Restore state after failed dispute requests

dechov committed:
Fix table heading

Evidence stylesheet was accidentally affecting other screens to good effect. This change restores the appearance and further tweaks the styling.

vbelolapotkov reviewed (approved):
@dechov it looks good to me! 🎉 
I'd only ask you to add a comment/todo to the disputes styling.

dechov committed:
Add TODO calling out potentially unnecessary scoping of styling changes
